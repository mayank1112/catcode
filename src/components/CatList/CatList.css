.cats {
    display: grid;
    grid-template-columns: auto auto auto auto;
}

.cat {
    position: relative;
    width: 120px;
    height: 120px;
    text-align: center;
    margin: 4px;
}

.cat img {
    max-height: 100px;
}
  
.selectedCat::after, .cat:hover::after {
    position: absolute;
    left: 0;
    top: 0;
    background: #888;
    opacity: .4;
    width: 100%;
    height: 100%;
    content: '';
    border-radius: 24px;
}
  
.groupedCat {
    cursor: not-allowed;
}
