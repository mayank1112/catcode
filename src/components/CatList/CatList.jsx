import { CircularProgress } from "@material-ui/core";
import CatImage from "../shared/CatImage/CatImage";
import './CatList.css';

const CatList = ({ groupedCats, selectCat, selectedCats, cats }) => (
    <div className="cats">
        {cats && cats.length ? cats.map(cat => {
            const catId = cat.join('');
            let className = 'cat';
            if (groupedCats && groupedCats.includes(catId)) {
                className += ' groupedCat';
            } else if (selectedCats && selectedCats.includes(catId)) {
                className += ' selectedCat';
            }
            return (
                <button key={catId} className={className} onClick={() => selectCat(catId)}>
                    <CatImage catId={catId} />
                </button>
            )
        }) : <CircularProgress />
        }
    </div>
);

export default CatList;
