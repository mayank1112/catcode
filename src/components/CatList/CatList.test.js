import Enzyme, { shallow } from 'enzyme';
import CatList from './CatList';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

it('<CatList> component should match snapshot', () => {
    const component = shallow(<CatList />);
    expect(toJson(component)).toMatchSnapshot();
});
