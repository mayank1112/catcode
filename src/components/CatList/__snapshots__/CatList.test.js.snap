// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CatList> component should match snapshot 1`] = `
<div
  className="cats"
>
  <WithStyles(ForwardRef(CircularProgress)) />
</div>
`;
